core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java

```diff
@@ -50,10 +53,23 @@
      * <p> The compiled patterns and their associated target objects </p>
      */
     List<Mapping<E>> compiledPatterns = new ArrayList<>();
-    ;
+
+    /**
+     * This flag controls if passed named params should be appended
+     * to the map in {@link #replaceParameters(Map, Map)}
+     * and will be accessible in {@link com.opensymphony.xwork2.config.entities.ResultConfig}.
+     * If set to false, the named parameters won't be appended.
+     *
+     * This behaviour is controlled by {@link org.apache.struts2.StrutsConstants#STRUTS_MATCHER_APPEND_NAMED_PARAMETERS}
+     *
+     * @since 2.5.23
+     * See WW-5065
+     */
+    private final boolean appendNamedParameters;
     
-    public AbstractMatcher(PatternMatcher<?> helper) {
+    public AbstractMatcher(PatternMatcher<?> helper, boolean appendNamedParameters) {
```

Comment:
Because the original method 
```
public AbstractMatcher(PatternMatcher<?> helper)
```
is public, people may have extended this class, which would then break their code unexpectedly with this change.  Maybe it would be a good idea to add keep the old method signature present with something like:
```
public AbstractMatcher(PatternMatcher<?> helper) {
    this(helper, true);  // Set boolean to whatever default setting is decided on
}
```
That should preserve backward-compatbility.

Reply:
Right, done!